invoke/runners.py

```diff
@@ -322,8 +322,22 @@ def get():
                     # Can't use six.b because that just assumes latin-1 :(
                     data = data.encode(self.encoding)
                 yield data
-        # Decode stream using our generator & requested encoding
-        for data in codecs.iterdecode(get(), self.encoding, errors='replace'):
+        if not hasattr(output, 'encoding') or self.encoding == output.encoding:
+            # No need in re-encoding if output is a BytesIO or CarbonCopy, or
+            # source and target encodings are the same
+            data_stream = get()
+        else:
+            # Re-encode from user requested encoding to the output encoding
+            data_stream = codecs.iterencode(
+                codecs.iterdecode(
+                    get(),
+                    self.encoding,
+                    errors='replace'
+                ),
+                output.encoding,
+                errors='replace'
+            )
+        for data in data_stream:
```

Comment:
1. `get` always yields `bytes` (`str` in Python 2.x)
2. `encode(decode(bytes))` returns `bytes`
3. `output.write` works safer with `bytes`

I don't see any mistake here.